lib/taskana-core-test/src/test/java/acceptance/task/create/CreateTaskAccTest.java

```diff
@@ -0,0 +1,967 @@
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test //
+  void should_ThrowException_When_AttachmentIsInvalid() throws Exception {
+    Consumer<Attachment> testCreateTask =
+        invalidAttachment -> {
+          Task taskWithInvalidAttachment = createDefaultTask();
+          taskWithInvalidAttachment.addAttachment(invalidAttachment);
+          ThrowingCallable call = () -> taskService.createTask(taskWithInvalidAttachment);
+          assertThatThrownBy(call)
+              .describedAs(
+                  "Should have thrown an InvalidArgumentException, "
+                      + "because Attachment-ObjRef is null.")
+              .isInstanceOf(InvalidArgumentException.class);
+        };
+
+    defaultAttachment.setObjectReference(null);
+    testCreateTask.accept(defaultAttachment);
+
+    ObjectReference objectReference =
+        ObjectReferenceBuilder.newObjectReference()
+            .system("SYSTEM_B")
+            .systemInstance("INSTANCE_B")
+            .company("COMPANY_A")
+            .type("ArchiveId")
+            .value(null)
+            .build();
+    defaultAttachment.setObjectReference(objectReference);
+    testCreateTask.accept(defaultAttachment);
+
+    ObjectReference objectReference1 =
+        ObjectReferenceBuilder.newObjectReference()
+            .system("SYSTEM_B")
+            .systemInstance("INSTANCE_B")
+            .company("COMPANY_A")
+            .type(null)
+            .value("12345678901234567890123456789012345678901234567890")
+            .build();
+    defaultAttachment.setObjectReference(objectReference1);
+    testCreateTask.accept(defaultAttachment);
+
+    ObjectReference objectReference2 =
+        ObjectReferenceBuilder.newObjectReference()
+            .system("SYSTEM_B")
+            .systemInstance("INSTANCE_B")
+            .company(null)
+            .type("ArchiveId")
+            .value("12345678901234567890123456789012345678901234567890")
+            .build();
+    defaultAttachment.setObjectReference(objectReference2);
+    testCreateTask.accept(defaultAttachment);
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UseCustomName_For_NewTask() throws Exception {
+    Task task = createDefaultTask();
+    task.setName("Test Name");
+
+    Task createdTask = taskService.createTask(task);
+
+    assertThat(createdTask).isNotNull();
+    assertThat(createdTask.getCreator()).isEqualTo("user-1-2");
+    assertThat(createdTask.getName()).isEqualTo("Test Name");
+  }
+
+  @WithAccessId(user = "user-1-2")
+  @Test
+  void should_UseClassificationMetadataFromCorrectDomain_For_NewTask() throws Exception {
+    ClassificationSummary classificationSummary =
+        ClassificationBuilder.newClassification()
+            .key(defaultClassificationSummary.getKey())
+            .priority(99)
+            .domain("DOMAIN_B")
+            .buildAndStoreAsSummary(classificationService, "admin");
+    WorkbasketSummary workbasketSummary =
```

Comment:
Because we want to make a case where we have 2 classifications with the same key but different domain, one with "DOMAIN_A" and one with "DOMAIN_B", and then make sure if the classification metadata from the correct Domain (DOMAIN_B) is fetched.  In the taskService.createTask(), we query the classification using the parameters Classification Key and workBasket Domain. Therefore we have to assign a task to a workbasket with Domain "DOMAIN_B". Hence is a new workbasket with "DOMAIN_B"

Reply:
In this case, additional Classification is enough. You don't need additional Workbasket, because you just replace the defaultWorkbasket with your new Workbasket in this test. As the defaultWorkbasket has "DOMAIN_A", you check if the Classification Domain is the same. Because you create a Classification with "DOMAIN_B" in this test, the check for the domain of Classification is relevant, as the Classification could have "DOMAIN_A" or "DOMAIN_B"